### PR TITLE
Fetch version info from publish API

### LINKED ISSUE #342: Fetch data from the publish prototype to the GUI
Part of #340 
Hook up the publish API into the GUI so that it is possible to fetch the current version data. Displaying it cleanly is out of scope.

### PR DESCRIPTION
**Pull request by dchiquit**
Define a new axios client for using the publish API.

Define a new `publish` store for storing stuff from the publish API,
much as we have a `girder` store to store stuff from girder.

Display the published versions on the DandisetLandingView. This display
code is temporary to prove that the underlying code works as intended.

Fixes #342

### CODE REVIEW

File: web/src/rest.js [hunk 1]
```diff
@@ -1,14 +1,19 @@
 
+import axios from 'axios';
 import { RestClient } from '@girder/components/src';
 
 const apiRoot = process.env.VUE_APP_API_ROOT || 'http://localhost:8080/api/v1';
+const publishApiRoot = process.env.PUBLISH_APP_API_ROOT || 'http://localhost:8000/api';
```
Comment (brianhelba) [inline web/src/rest.js:6]: The `VUE_APP_` prefix is magic, and is necessary for the value to be injected: https://cli.vuejs.org/guide/mode-and-env.html#environment-variables

Perhaps `VUE_APP_PUBLISH_API_ROOT`?

Comment (brianhelba) [inline web/src/rest.js:6]: I think it'd be better to to define a typical development default of the publish API root environment variable in `.env.development`, and omit the default value here (likewise with `apiRoot`). For production builds, I don't think we want this fallback default, as we want the app to fail obviously if the environment was not configured.

File: web/src/views/DandisetLandingView/DandisetLandingView.vue [hunk 2]
```diff
@@ -270,6 +277,10 @@ export default {
     ...mapState('girder', {
       selected: (state) => state.currentDandiset,
     }),
+    ...mapState('publish', {
+      mostRecentDandiset: (state) => state.mostRecentDandiset,
+      versions: (state) => state.versions,
+    }),
```
Comment (brianhelba) [inline web/src/views/DandisetLandingView/DandisetLandingView.vue:283]: This can be simplified: https://vuex.vuejs.org/guide/state.html#the-mapstate-helper

```suggestion
    ...mapState('publish', [
      'mostRecentDandiset',
      'versions',
    ]),
```

File: web/src/store/publish.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  mutations: {
+    setVersions(state, versions) {
+      state.versions = versions;
+    },
+  },
+  actions: {
+    async fetchPublishedVersions({ commit }, identifier) {
+      if (identifier) {
+        const response = await publishRest.get(`dandisets/${identifier}/versions/`)
+          .catch((error) => {
```
Comment (brianhelba) [inline web/src/store/publish.js:17]: I think using `try...catch` exclusively with `await` is cleaner in most cases, including here.

File: web/src/store/publish.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { publishRest } from '@/rest';
+
+export default {
+  namespaced: true,
+  state: {
+    versions: null,
```
Comment (brianhelba) [inline web/src/store/publish.js:6]: I don't think any of this should be in Vuex at all. Since every different navigation of the `DandisetLandingView` is going to be overwriting all of the state here, it's better to keep it within the `DandisetLandingView` itself, rather than deal with the extra complexity and potential desynchronization bugs of having it in a singleton Vuex module.

File: web/src/views/DandisetLandingView/DandisetLandingView.vue [hunk 5]
```diff
@@ -301,6 +312,10 @@ export default {
         }
       },
     },
+    // TODO move this somewhere more sensical
+    async meta(val) {
+      this.$store.dispatch('publish/fetchPublishedVersions', val.identifier);
```
Comment (brianhelba) [inline web/src/views/DandisetLandingView/DandisetLandingView.vue:317]: How about putting this at the end of the `id` watcher, which initializes this component on creation or navigation?

File: web/src/views/DandisetLandingView/DandisetLandingView.vue [hunk 6]
```diff
@@ -270,6 +277,10 @@ export default {
     ...mapState('girder', {
       selected: (state) => state.currentDandiset,
     }),
+    ...mapState('publish', {
+      mostRecentDandiset: (state) => state.mostRecentDandiset,
```
Comment (AlmightyYakob) [inline web/src/views/DandisetLandingView/DandisetLandingView.vue:281]: I don't see where this exists in the store, and it doesn't look like it's used anywhere else in this component either. Why is this here?

File: web/src/store/publish.js (hunk 4 above)
Comment (AlmightyYakob) [reply web/src/store/publish.js:6]: I agree, and if `DandisetLandingView` is the only place using this it makes sense. I know we plan on showing version info in the main dandiset listing component ([figma link](https://www.figma.com/proto/kEHOgqU0rcOIyCYIDlpSMV/DANDI-UI?scaling=min-zoom&node-id=16%3A78)), but that seems orthogonal to this, and depends on the specifics of how we want display that info.

File: web/src/views/DandisetLandingView/DandisetLandingView.vue (hunk 6 above)
Comment (dchiquit) [reply web/src/views/DandisetLandingView/DandisetLandingView.vue:281]: I thought it might be useful, but decided it was too much for this review. I forgot to remove it here, my b

File: web/src/rest.js [hunk 7]
```diff
@@ -3,7 +3,7 @@ import axios from 'axios';
 import { RestClient } from '@girder/components/src';
 
 const apiRoot = process.env.VUE_APP_API_ROOT || 'http://localhost:8080/api/v1';
-const publishApiRoot = process.env.PUBLISH_APP_API_ROOT || 'http://localhost:8000/api';
```
Comment (brianhelba) [inline web/src/rest.js]: Can you remove this default too, just for consistency?
